uat/custom-components/src/main/java/com/aws/greengrass/utils/LoggingHelper.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.utils;
+
+import ch.qos.logback.classic.Logger;
+
+import java.util.concurrent.TimeUnit;
+
+public final class LoggingHelper {
+    private static double random;
+    public LoggingHelper() {
+    }
+
+    public static void performLogging(Logger logger, int totalLogNumbers, long logWriteFreqSeconds, int logMsgSizeBytes) {
+        for (int i=0; i < totalLogNumbers; i++) {
+            random = Math.random();
+            StringBuilder sb = generateLogMessage(i, logMsgSizeBytes);
+            logger.info(sb.toString());
+            try {
+                TimeUnit.SECONDS.sleep(logWriteFreqSeconds);
+            } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)

Reply:
We should not just print if someone interrupts then how will this exit upon interruption. We should maybe use `System.exit(0)`